cpp11/Ice/hello/Server.cpp

```diff
@@ -48,11 +46,13 @@ int main(int argc, char* argv[])
         else
         {
             auto adapter = communicator->createObjectAdapter("Hello");
-            adapter->add(make_shared<HelloI>(), Ice::stringToIdentity("hello"));
+            adapter->add(make_shared<HelloI>(requestShutdown), Ice::stringToIdentity("hello"));
             adapter->activate();
 
-            communicator->waitForShutdown();
+            // The shutdown can be requested by Hello::shutdown or the CtrlCHandler.
+            shutdownRequested.wait();
         }
+        communicator->shutdown();
```

Comment:
Keep in mind `Communicator::shutdown` means "initiate shutdown". You need to call waitForShutdown if you want to wait until the shutdown is actually complete.